src/main/java/org/jitsi/videobridge/EncodingsManager.java

```diff
@@ -58,7 +57,7 @@
      * @param secondarySsrc the secondary SSRC in the SSRC association.
      * @param type the association type.
      */
-    public void addSsrcAssociation(
+    public synchronized void addSsrcAssociation(
```

Comment:
I think we should release the lock before calling into the listeners (since we don't know what the code they run and they could be aqcuiring other locks). Also, as a general good practice synchronize on another object which is private to the instance.

Reply:
i'll take care of that, thanks

Reply:
i provided an alternative, please check